src/main/java/com/digitalpebble/storm/crawler/bolt/ParserBolt.java

```diff
@@ -152,21 +148,15 @@ public void prepare(Map conf, TopologyContext context,
 
         this.collector = collector;
 
-        this.eventMeters = context.registerMetric("parser-meter",
-                new MeterMetric(), 5);
-        this.eventTimers = context.registerMetric("parser-timer",
-                new TimerMetric(), 5);
-        this.eventHistograms = context.registerMetric("parser-histograms",
-                new HistogramMetric(), 5);
-
+        this.eventCounter = context.registerMetric(this.getClass()
+                .getSimpleName(), new MultiCountMetric(), 10);
```

Comment:
The component ID is already handled by the metrics mechanism and will be set for the datapoints.  Specifying the class here allows to add more info on which class was actually used, e.g. various flavours of parsers all running as 'parse' component.

Here is what you'd get by using the component name. Bit redundant, isn't it?

```
Mint15-Vostro-410:1027	  6:parse      	parse                  	{outlink_kept=952, tuple_in=14, error_out_link_parsing_MalformedURLException=105, tuple_success=14, outlink_filtered=119, outlink_outsideSourceDomainOrHostname=217}
```